src/Microsoft.AspNet.Razor/Generator/Compiler/CodeTree/Chunks/TagHelpers/TagHelperChunk.cs

```diff
@@ -29,5 +29,10 @@ public class TagHelperChunk : ChunkBlock
         /// The HTML tag name.
         /// </summary>
         public string TagName { get; set; }
+
+        /// <summary>
+        /// Indicates whether or not the tag is self closing.
+        /// </summary>
+        public bool SelfClosing { get; set; }
```

Comment:
ick; but :ok: drop the requirement for the plain POCO objects but impose it where there's already an explicit constructor